### ISSUE TITLE
Confusion about Num_classes and No-object

### ISSUE THREAD

**Issue by franziska-denk** [qianyuzqy/TransVOD_plusplus#8]
Hello @qianyuzqy !

First of all, thank you for publishing your nice work! :)

I am struggling to understand how the no-object class is handled in TransVOD. I was reading the paper and going through the code, however, I still couldn't figure it out, and I am sorry if I missed the explanation somewhere! But, here is what I am unsure about:

As far as I see and in accordance with [the logic in the DETR code](https://github.com/facebookresearch/detr/issues/108#issuecomment-650269223), you have `num_classes=31`, which is the number of classes in ImageNet VID dataset + 1, which corresponds to the no-object class.
Then, the logits of the class embedding layer are of size 31 (so index 0-30) in the last dimension, according to:
https://github.com/qianyuzqy/TransVOD_plusplus/blob/2f0b7cd17ce1bc88249f90545a7a2c7467a7c9ec/models/deformable_detr_multi.py#L54

 Assuming now category IDs starting from 1, as I could also see in [the datafiles that are provided for download](https://drive.google.com/drive/folders/1cCXY41IFsLT-P06xlPAGptG7sc-zmGKF), this means that the last number in the class logit vector does not correspond to the no-object class, but the class with ID 30.

In the class loss calculation, I can see that the targets for non-objects are filled with `num_classes`, which is 31:
https://github.com/qianyuzqy/TransVOD_plusplus/blob/2f0b7cd17ce1bc88249f90545a7a2c7467a7c9ec/models/deformable_detr_multi.py#L239-L241
Directly afterwards, the target classes are one-hot encoded and only dimensions 0-30 are used to calculate the loss for the classification logits (which makes sense to me, as labels and logits should have the same shape):
https://github.com/qianyuzqy/TransVOD_plusplus/blob/2f0b7cd17ce1bc88249f90545a7a2c7467a7c9ec/models/deformable_detr_multi.py#L243-L248

But that means also that for queries that correspond to no-object, there is no ground truth label (the whole target class vector is 0, as the 1 was in the 31th dimension, which was removed by the slicing operation). From what I saw, this has the effect during training that all class logits of no-object queries are learned to be very low numbers, as there is no ground truth to match.

I am wondering why this is done like that? Why is the concept of DETR with the explicit no-object class in the class logits not followed? Is this different approach needed because of the confidence sorting stage of the object queries after the spatial transformers?

I would highly appreciate it if you could let me know what I am missing so that I can fully understand the approach! :) thank you!
